src/modm/platform/core/stm32/module.lb

```diff
@@ -57,6 +57,7 @@ def build(env):
     # (cycles per loop, setup cost in loops)
     tuning = {
         "g4": (3,  4), # CM4  tested on G476 in RAM
+        "c0": (3,  4), # CM0+ TODO: Test
```

Comment:
TODO?

Reply:
Yes, AFAIR that comment was from me. Did not know how to test :-/

Reply:
Ah, sorry it's [the examples/generic/delay code]() and then you would need to run this through the code in [this blog post](https://blog.salkinium.com/modm-delay/).

I've just done this on the Nucleo-C031C6 and the delay is comparable to other CM0(+) devices:

| Device | Core Type | Cycles per Loop | Minimum Cycles at Boot/High Frequency | Minimum Delay at Boot Frequency | Minimum Delay at High Frequency |
|:-----------|:-----|:---:|:-----:|--------------------:|------------------:|
| **STM32C031**  | **cm0+** |  **3**  | **15/17** | **1250ns @ 12     MHz** | **354ns @  48 MHz**   |
| STM32F072  | cm0  |  4  | 18/19 | 1125ns @ 16     MHz | 395ns @  48 MHz   |
| STM32F091  | cm0  |  4  | 18/19 | 1125ns @ 16     MHz | 395ns @  48 MHz   ||
| STM32G071  | cm0+ |  3  | 16/18 | 1000ns @ 16     MHz | 281ns @  64 MHz   |
| STM32L031  | cm0  |  3  | 16/17 | 7629ns @  2.097 MHz | 531ns @  32 MHz   |

~1.25us minimum delay on boot frequency is almost on the ideal line:

![](https://github.com/user-attachments/assets/46ac4f79-f3c8-469a-afea-323bafd5a2cb)

~400ns minimum delay at 48MHz:

![](https://github.com/user-attachments/assets/f7d71fc8-044e-408e-89bf-7c3e651d4330)

Also no scaling error for longer delays:

![](https://github.com/user-attachments/assets/a6ef289a-4980-4858-be6e-1a5453117dd1)